ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutor.java

```diff
@@ -303,6 +303,11 @@ private void terminateQuery(final PreparedStatement<TerminateQuery> terminateQue
 
     final Optional<PersistentQueryMetadata> query = ksqlEngine.getPersistentQuery(queryId.get());
     query.ifPresent(PersistentQueryMetadata::close);
+
+    if (!query.isPresent()) {
```

Comment:
I believe this is invoked from reading the command from the command topic, which we're not planning on doing, so this shouldn't be required.